transaction/simple.go

```diff
@@ -2,29 +2,35 @@ package transaction
 
 import (
 	"github.com/gluster/glusterd2/context"
+
+	"github.com/pborman/uuid"
 )
 
 // SimpleTxn is transaction with fixed stage, commit and store steps
 type SimpleTxn struct {
 	// Ctx is the transaction context
 	Ctx *context.Context
 	// Nodes are the nodes where the stage and commit functions are performed
-	Nodes []string
+	Nodes []uuid.UUID
 	// LockKey is the key to be locked
 	LockKey string
```

Comment:
Where does the locking begin and end ? IOW, at what step is the lock taken and then relinquished ?